src/Data/Tree/Binary.agda

```diff
@@ -10,56 +10,76 @@ module Data.Tree.Binary where
 
 open import Level using (Level; _⊔_)
 open import Data.List.Base using (List)
-open import Data.DifferenceList as DiffList using (DiffList; []; _∷_; _∷ʳ_; _++_)
+open import Data.DifferenceList as DiffList using (DiffList; []; _∷_; _∷ʳ_; _++_; [_])
 open import Data.Nat.Base using (ℕ; zero; suc; _+_)
 open import Function.Base
-open import Relation.Binary.PropositionalEquality
 
 private
   variable
-    a b : Level
+    a b c d : Level
     A : Set a
     B : Set b
+    C : Set c
+    D : Set d
 
-data Tree (A : Set a) : Set a where
-  leaf : Tree A
-  node : Tree A → A → Tree A → Tree A
+data Tree (A : Set a) (B : Set b) : Set (a ⊔ b) where
```

Comment:
We should either add a comment explaining which type is which, or we should name types `A` and `B` something a little more meaningful?

Reply:
I'll do both